### PR TITLE
Added 3 way merge sort

### LINKED ISSUE #660: Three Way Merge Sort
I can implement Three Way Merge Sort in C++ as part of GSSOC'21. Kindly assign this issue to me

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
Fixes #660 Kindly review this @OjusWiZard @rudrakshi99 @tarun26091999 @plazzy99

### CODE REVIEW

File: Code/C++/three_way_merge_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
+// C++ program to implement Three Way Merge Sort
+
```
Comment (rudrakshi99) [inline Code/C++/three_way_merge_sort.cpp:2]: add a description to the algorithm

File: Code/C++/three_way_merge_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        cout<<temp[i]<<" ";
+    cout<<endl;
+
+    return 0;
+}
+
+/*
+
+Time Complexity: O(n*logn)
+Space Complexity: O(n)
```
Comment (rudrakshi99) [inline Code/C++/three_way_merge_sort.cpp:159]: add complexities at the end.

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

File: Code/C++/three_way_merge_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+SAMPLE INPUT AND OUTPUT
+
```
Comment (rudrakshi99) [inline Code/C++/three_way_merge_sort.cpp:162]: write `Test cases :` instead of this.

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

**Comment by rudrakshi99** [comment]
Please resolve merge conflicts.

**Comment by Harikrishnan6336** [comment]
@rudrakshi99 Done.

**Comment by plazzy99** [comment]
Resolve conflicts

**Comment by Harikrishnan6336** [comment]
> Resolve conflicts

Done.

File: Algorithm/Searching_Sorting/readme.md [hunk 4]
```diff
@@ -77,9 +77,9 @@ Following are the types of searches which we will be discussing in this book.
 * Book Allocation ----> [Java](/Code/Java/Book_Allocation.java)
 * Number of times sorted array rotated ---->[C++](/Code/C++/no_of_rotation.cpp)
 * Staircase Search ----> [C++](/Code/C++/staircase_search.cpp)
+* Three Way Merge Sort ----> [C++](/Code/C++/three_way_merge_sort.cpp)
 * Searching in sorted rotated array ----> [Java](/Code/Java/searching_in_sorted_rotated_array.java)
```
Comment (krishna-NIT) [inline Algorithm/Searching_Sorting/readme.md:81]: @Harikrishnan6336  Arrange the question in alphabetical order

File: Code/C++/three_way_merge_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+The numbers in sorted order is:  1  2  3  4  5
+
+2.
+Input - 
+How many numbers do you want to sort? 0
+Output - 
+There are no numbers to sort!!!
+
+Time Complexity: O(n*logn)
+Space Complexity: O(n)
```
Comment (krishna-NIT) [inline Code/C++/three_way_merge_sort.cpp:176]: add what is n here?

File: Code/C++/three_way_merge_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        cout<<temp[i]<<" ";
+    cout<<endl;
+
+    return 0;
+}
+
+/*
+Test Cases
+1.
+Input -
```
Comment (krishna-NIT) [inline Code/C++/three_way_merge_sort.cpp:163]: give proper space and use tabs properly for making it more readable

Comment (krishna-NIT) [reply Code/C++/three_way_merge_sort.cpp]: do this @Harikrishnan6336

File: Algorithm/Searching_Sorting/readme.md [hunk 7]
```diff
@@ -77,9 +77,9 @@ Following are the types of searches which we will be discussing in this book.
...
 * Find peak element using binary search ----> [Python](/Code/Python/peak_element.py)
 * Cycle Sort ----> [C++](/Code/C++/cycle_sort.cpp)
```
Comment (krishna-NIT) [inline Algorithm/Searching_Sorting/readme.md:83]: Rearrange all questions in alphabetical order

Comment (Harikrishnan6336) [reply Algorithm/Searching_Sorting/readme.md]: Done.

**Comment by krishna-NIT** [comment]
@Harikrishnan6336  Good Efforts ,
@rish-singhal LGTM

**Pull request merged**